java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/protocol/PullUpRefactoring.java

```diff
@@ -0,0 +1,280 @@
+    }
+
+    private List<QuickPickItem> getMembers(NbCodeLanguageClient client, String uri, int offset, QuickPickItem source, QuickPickItem target) {
+        List<QuickPickItem> members = new ArrayList<>();
+        try {
+            FileObject file = Utils.fromUri(uri);
+            JavaSource js = JavaSource.forFileObject(file);
+            if (js == null) {
+                throw new IOException("Cannot get JavaSource for: " + uri);
+            }
+            js.runUserActionTask(info -> {
+                Trees trees = info.getTrees();
+                SourcePositions sourcePositions = trees.getSourcePositions();
+                ElementUtilities eu = info.getElementUtilities();
+                Types types = info.getTypes();
+                TypeElement sourceElement = (TypeElement) gson.fromJson(gson.toJson(source.getUserData()), ElementData.class).toHandle().resolve(info);
+                TypeMirror sourceType = sourceElement.asType();
+                TypeElement targetElement = (TypeElement) gson.fromJson(gson.toJson(target.getUserData()), ElementData.class).toHandle().resolve(info);
+                for (Element e : sourceElement.getEnclosedElements()) {
+                    switch (e.getKind()) {
+                        case CONSTRUCTOR:
+                        case STATIC_INIT:
+                        case INSTANCE_INIT:
+                            continue;
+                        case METHOD:
+                            if (eu.alreadyDefinedIn(e.getSimpleName(), (ExecutableType) types.asMemberOf((DeclaredType) sourceType, e), targetElement)) {
+                                break;
+                            }
+                        default: {
+                            TreePath path = trees.getPath(e);
+                            long startMember = path != null ? sourcePositions.getStartPosition(path.getCompilationUnit(), path.getLeaf()) : -1;
+                            long endMember = path != null ? sourcePositions.getEndPosition(path.getCompilationUnit(), path.getLeaf()) : -1;
+                            boolean selected = offset > startMember && offset < endMember;
+                            members.add(new QuickPickItem(createLabel(info, e), null, null, selected, new ElementData(e)));
+                        }
+                    }
+                }
+            }, true);
+        } catch (Exception ex) {
+            client.showMessage(new MessageParams(MessageType.Error, ex.getLocalizedMessage()));
+        }
+        return members;
+    }
+
+    private void pullUp(NbCodeLanguageClient client, String uri, QuickPickItem source, QuickPickItem target, List<QuickPickItem> members) {
+        try {
+            FileObject file = Utils.fromUri(uri);
+            ClasspathInfo info = ClasspathInfo.create(file);
+            JavaSource js = JavaSource.forFileObject(file);
+            if (js == null) {
+                throw new IOException("Cannot get JavaSource for: " + uri);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Exception [IOException] is explicitly thrown on this line but is caught in the outer catch clause with generic exception on line [240].  Consider adding a catch clause of this specific exception in the outer try statement.